<a href="https://colab.research.google.com/github/BelikMary/individ-zadanie/blob/main/Individyal.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### Импорт библиотек

In [1]:
import numpy as np
import pandas as pd

In [101]:
from matplotlib import pyplot as plt
%matplotlib inline
import warnings
warnings.filterwarnings('ignore')
import seaborn as sns

### Загрузка и предварительное изучение файла

In [13]:
from google.colab import files
uploaded = files.upload()

Saving list-of-airports-in-russian-federation-hxl-tags-1.csv to list-of-airports-in-russian-federation-hxl-tags-1.csv


In [163]:
traffic = pd.read_csv("russian_passenger_air_service_2.csv")

In [164]:
traffic.sample(n=15)

Airport name  Year  January  February    March    April  \
3308                   Tambov  2009      0.0       0.0      0.0      0.0   
3301                 Samsearn  2009      0.0       0.0      0.0      0.0   
2432                   Tambov  2012    293.0      52.0      0.0    580.0   
607                   Nalchik  2018      0.0       0.0      0.0      0.0   
3003                  Solovki  2010      0.0       0.0      0.0      0.0   
2619           Mezhdurechensk  2011      0.0       0.0      0.0      0.0   
2689    Rogachevo (Amderma-2)  2011      0.0       0.0      0.0      0.0   
3702                 Vanavara  2007      0.0       0.0      0.0      0.0   
2648             Novokuznetsk  2011   8010.0    7606.0   8764.0   6935.0   
2500                 Yartsevo  2012      0.0       0.0      0.0      0.0   
3579            North-Yenisei  2008      0.0       0.0      0.0      0.0   
2703            North-Yenisei  2011      0.0       0.0      0.0      0.0   
2700        Saratov (central)  2011  13003.0   14022.0  16204.0  18530.0   
1801  Gerbil (Kolguev Island)  2014      0.0       0.0      0.0      0.0   
1549                 Samsearn  2015      0.0       0.0      0.0      0.0   

          May     June     July   August  September  October  November  \
3308      0.0      0.0      0.0      0.0        0.0      0.0       0.0   
3301      0.0      0.0      0.0      0.0        0.0      0.0       0.0   
2432   1343.0   1460.0   1898.0   2073.0     1617.0   1864.0    1564.0   
607       0.0      0.0      0.0      0.0        0.0      0.0       0.0   
3003      0.0      0.0      0.0      0.0        0.0      0.0       0.0   
2619      0.0      0.0      0.0      0.0        0.0      0.0       0.0   
2689      0.0      0.0      0.0      0.0        0.0      0.0       0.0   
3702      0.0      0.0      0.0      0.0        0.0      0.0       0.0   
2648   6530.0  10483.0  14003.0  14100.0    10121.0   8333.0    7012.0   
2500      0.0      0.0      0.0      0.0        0.0      0.0       0.0   
3579      0.0      0.0      0.0      0.0        0.0      0.0       0.0   
2703      0.0      0.0      0.0      0.0        0.0      0.0       0.0   
2700  20945.0  26623.0  29977.0  30551.0    26562.0  19839.0   18089.0   
1801      0.0      0.0      0.0      0.0        0.0      0.0       0.0   
1549      0.0      0.0      0.0      0.0        0.0      0.0       0.0   

      December  Whole year                            Airport coordinates  
3308       0.0         0.0   (Decimal('41.491404'), Decimal('52.803629'))  
3301       0.0         0.0  (Decimal('117.640738'), Decimal('62.183775'))  
2432       0.0     14315.0   (Decimal('41.491404'), Decimal('52.803629'))  
607        0.0         0.0   (Decimal('43.635863'), Decimal('43.508409'))  
3003       0.0         0.0   (Decimal('35.723241'), Decimal('65.029939'))  
2619       0.0         0.0                                      Not found  
2689       0.0         0.0   (Decimal('52.477414'), Decimal('71.609701'))  
3702       0.0         0.0  (Decimal('102.303298'), Decimal('60.351026'))  
2648     162.0    110246.0   (Decimal('86.864303'), Decimal('53.805377'))  
2500       0.0         0.0   (Decimal('90.203907'), Decimal('60.252931'))  
3579       0.0         0.0   (Decimal('93.016353'), Decimal('60.373043'))  
2703       0.0         0.0   (Decimal('93.016353'), Decimal('60.373043'))  
2700      58.0    251423.0  (Decimal('106.644236'), Decimal('47.454355'))  
1801       0.0         0.0   (Decimal('50.249107'), Decimal('69.186265'))  
1549       0.0         0.0  (Decimal('117.640738'), Decimal('62.183775'))

In [165]:
traffic.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3961 entries, 0 to 3960
Data columns (total 16 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Airport name         3961 non-null   object 
 1   Year                 3961 non-null   int64  
 2   January              3961 non-null   float64
 3   February             3961 non-null   float64
 4   March                3961 non-null   float64
 5   April                3961 non-null   float64
 6   May                  3961 non-null   float64
 7   June                 3961 non-null   float64
 8   July                 3961 non-null   float64
 9   August               3961 non-null   float64
 10  September            3961 non-null   float64
 11  October              3961 non-null   float64
 12  November             3961 non-null   float64
 13  December             3961 non-null   float64
 14  Whole year           3961 non-null   float64
 15  Airport coordinates  3961 non-null   o

In [185]:
traffic.drop(['Airport coordinates'],axis=1,inplace=True)

In [186]:
traffic.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3961 entries, 0 to 3960
Data columns (total 16 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Airport name     3961 non-null   object 
 1   Year             3961 non-null   int64  
 2   January          3961 non-null   float64
 3   February         3961 non-null   float64
 4   March            3961 non-null   float64
 5   April            3961 non-null   float64
 6   May              3961 non-null   float64
 7   June             3961 non-null   float64
 8   July             3961 non-null   float64
 9   August           3961 non-null   float64
 10  September        3961 non-null   float64
 11  October          3961 non-null   float64
 12  November         3961 non-null   float64
 13  December         3961 non-null   float64
 14  Whole year       3961 non-null   float64
 15  True whole year  3961 non-null   float64
dtypes: float64(14), int64(1), object(1)
memory usage: 495.2+ KB


In [187]:
traffic.describe(include=['O'])

Airport name
count          3961
unique          298
top          Usinsk
freq             28

### Понаблюдаем и сделаем вывод, что всего у нас представлено 298 аэропортов. Найдем топ-100 аэропортов с самым высоким пассажирским трафиком.

In [188]:
traffic.head()

Airport name  Year  January  February  March  April  May  June  July  \
0           Abakan  2020  13495.0   14940.0    0.0    0.0  0.0   0.0   0.0   
1           Aikhal  2020      0.0       0.0    0.0    0.0  0.0   0.0   0.0   
2             Loss  2020      0.0       0.0    0.0    0.0  0.0   0.0   0.0   
3          Amderma  2020      0.0       0.0    0.0    0.0  0.0   0.0   0.0   
4  Anadyr (Carbon)  2020   4255.0    4565.0    0.0    0.0  0.0   0.0   0.0   

   August  September  October  November  December  Whole year  True whole year  
0     0.0        0.0      0.0       0.0       0.0         0.0          28435.0  
1     0.0        0.0      0.0       0.0       0.0         0.0              0.0  
2     0.0        0.0      0.0       0.0       0.0         0.0              0.0  
3     0.0        0.0      0.0       0.0       0.0         0.0              0.0  
4     0.0        0.0      0.0       0.0       0.0         0.0           8820.0

In [189]:
traffic['True whole year']=traffic['January'] + traffic['February'] + traffic['March'] + traffic['April'] + traffic['May']+ traffic['June'] + traffic['July'] + traffic['August'] + traffic['September'] + traffic['October'] + traffic['November'] + traffic['December']

In [214]:
traffic1=traffic.groupby(['Airport name'])['True whole year']. sum().reset_index()
traffic1=traffic1.sort_values(by='True whole year', ascending=False).head(100)
traffic1

Airport name  True whole year
118           Moscow (Sheremetyevo)     3.495603e+08
116             Moscow (Domodedovo)     3.256831e+08
119                Moscow (Vnukovo)     1.585543e+08
184      Saint Petersburg (Pulkovo)     1.487107e+08
44          Ekaterinburg (Koltsovo)     4.816369e+07
..                              ...              ...
154                        Omsukçan     1.106740e+04
53         Gulf Lawrence (Lawrence)     1.041500e+04
269                     Yoshkar-Ola     9.052000e+03
176  Providence (Providence Harbor)     8.160000e+03
288                         Çumikan     5.991350e+03

[100 rows x 2 columns]

In [216]:
df=traffic
o=traffic1['Airport name']
df=df.loc[df['Airport name']. isin(o)]
df

Airport name  Year  January  February    March    April  \
0                     Abakan  2020  13495.0   14940.0      0.0      0.0   
4            Anadyr (Carbon)  2020   4255.0    4565.0      0.0      0.0   
5          Anapa (Vitjazevo)  2020  43359.0   33653.0      0.0      0.0   
8       Arkhangelsk (Talagy)  2020  62698.0   61408.0      0.0      0.0   
9     Astrakhan (Narimanovo)  2020  47384.0   46387.0      0.0      0.0   
...                      ...   ...      ...       ...      ...      ...   
3950   Egvekinot (Bay Cross)  2007      0.0       0.0      0.0      0.0   
3955      Yuzhno-(Khomutovo)  2007  42441.0   35639.0  38866.0  42524.0   
3956                 Yakutsk  2007  27636.0   25976.0  34291.0  38803.0   
3958                 Yamburg  2007   6239.0    3195.0   6648.0   6268.0   
3959    Yaroslavl (Tunoshna)  2007    557.0     720.0    650.0    411.0   

          May     June     July   August  September  October  November  \
0         0.0      0.0      0.0      0.0        0.0      0.0       0.0   
4         0.0      0.0      0.0      0.0        0.0      0.0       0.0   
5         0.0      0.0      0.0      0.0        0.0      0.0       0.0   
8         0.0      0.0      0.0      0.0        0.0      0.0       0.0   
9         0.0      0.0      0.0      0.0        0.0      0.0       0.0   
...       ...      ...      ...      ...        ...      ...       ...   
3950      0.0      0.0      0.0      0.0        0.0      0.0       0.0   
3955  45660.0  57415.0  71830.0  74751.0    63180.0  56061.0   47176.0   
3956  38478.0  59142.0  67295.0  75257.0    52681.0  44505.0   40439.0   
3958   6088.0   6028.0   5975.0   6024.0     6262.0   6451.0    6968.0   
3959    328.0    864.0    688.0    965.0      883.0    505.0     640.0   

      December  Whole year  True whole year  
0         0.00         0.0         28435.00  
4         0.00         0.0          8820.00  
5         0.00         0.0         77012.00  
8         0.00         0.0        124106.00  
9         0.00         0.0         93771.00  
...        ...         ...              ...  
3950      0.00         0.0             0.00  
3955   1632.10    625294.0        577175.10  
3956   1916.60    545577.0        506419.60  
3958      8.87     72626.0         66154.87  
3959   2055.60      8277.0          9266.60  

[1326 rows x 16 columns]

In [217]:
df2 = df.melt(id_vars = ['Airport name', 'Year'],
         value_vars = ['January', 'February', 'March', 'April', 'May','June', 'July', 'August', 'September', 'October', 'November','December'])

df2.rename(columns = {'variable':'Month', 'value':'Traffic'}, inplace = True)

In [218]:
Months = ['January', 'February', 'March', 'April', 'May','June', 'July', 'August', 'September', 'October', 'November','December']
df2['Month'] = pd.Categorical(df2['Month'], categories=Months, ordered=True)
df2

Airport name  Year     Month   Traffic
0                      Abakan  2020   January  13495.00
1             Anadyr (Carbon)  2020   January   4255.00
2           Anapa (Vitjazevo)  2020   January  43359.00
3        Arkhangelsk (Talagy)  2020   January  62698.00
4      Astrakhan (Narimanovo)  2020   January  47384.00
...                       ...   ...       ...       ...
15907   Egvekinot (Bay Cross)  2007  December      0.00
15908      Yuzhno-(Khomutovo)  2007  December   1632.10
15909                 Yakutsk  2007  December   1916.60
15910                 Yamburg  2007  December      8.87
15911    Yaroslavl (Tunoshna)  2007  December   2055.60

[15912 rows x 4 columns]

Сводная таблица позволяет отследить тенденцию изменения пассажирских поток в отдельных аэропортах (визуализация представлена на Дашборде)

In [215]:
port_year = traffic.pivot_table(index='Airport name', columns='Year', values='True whole year')

port_year=port_year.join(traffic1.set_index('Airport name'), on='Airport name').sort_values(by='True whole year', ascending=False).head(12)
port_year

2007         2008         2009  \
Airport name                                                        
Moscow (Sheremetyevo)       12904794.00  14060490.30  13475813.00   
Moscow (Domodedovo)         17366204.99  19270075.20  17206576.74   
Moscow (Vnukovo)             6275435.56   7462158.84   7139655.45   
Saint Petersburg (Pulkovo)   5696180.00   6650112.00   6242703.00   
Ekaterinburg (Koltsovo)      2083193.80   2284352.97   1915862.90   
Sochi                        1530187.00   1526645.00   1300036.00   
Novosibirsk (Tolmachyovo)    1612308.10   1822688.80   1548111.80   
Krasnodar (Paszkowski)       1284941.00   1513055.00   1446645.00   
Ufa                          1038922.97   1214873.79   1086152.56   
Vladivostok (Knevichi)        854052.30    937777.30    909966.80   
Khabarovsk (New)              908867.66   1029806.63    954079.73   
Kaliningrad (Hrabrovo)        988291.67   1757848.84   1282661.75   

                                   2010         2011         2012  \
Airport name                                                        
Moscow (Sheremetyevo)       17757133.50  20657723.00  24099349.30   
Moscow (Domodedovo)         20652989.08  23880678.11  26108156.46   
Moscow (Vnukovo)             8818975.24   7611555.37   9021921.07   
Saint Petersburg (Pulkovo)   7822992.00   8925500.00  10403457.00   
Ekaterinburg (Koltsovo)      2483145.77   3033761.99   3460540.69   
Sochi                        1812837.00   1968383.00   1993483.00   
Novosibirsk (Tolmachyovo)    1962239.20   2403860.80   2867617.90   
Krasnodar (Paszkowski)       1936654.00   2329197.22   2409133.41   
Ufa                          1345427.44   1517476.27   1738523.08   
Vladivostok (Knevichi)       1168768.20   1326361.30   1478362.50   
Khabarovsk (New)             1260868.81   1396316.43   1636604.53   
Kaliningrad (Hrabrovo)        940500.23   1116480.45   1101569.28   

                                   2013         2014         2015  \
Airport name                                                        
Moscow (Sheremetyevo)       26898279.40  29073961.30  29004589.10   
Moscow (Domodedovo)         28719186.74  30945044.81  28599018.84   
Moscow (Vnukovo)            10291347.51  11760364.68  15082986.91   
Saint Petersburg (Pulkovo)  11998234.00  13335201.00  12711487.00   
Ekaterinburg (Koltsovo)      3940267.89   4175468.70   3904104.90   
Sochi                        2256543.27   2911447.53   3897592.85   
Novosibirsk (Tolmachyovo)    3308761.00   3548834.80   3343618.10   
Krasnodar (Paszkowski)       2650217.72   3197832.24   2910456.96   
Ufa                          2020521.85   2192404.88   2131378.93   
Vladivostok (Knevichi)       1696568.50   1643126.70   1558129.40   
Khabarovsk (New)             1828735.67   1787511.30   1603211.90   
Kaliningrad (Hrabrovo)       1278289.94   1354541.01   1427274.58   

                                   2016         2017         2018  \
Airport name                                                        
Moscow (Sheremetyevo)       30945912.40  36555045.10  41714366.30   
Moscow (Domodedovo)         26525585.67  28622313.97  27564916.81   
Moscow (Vnukovo)            12863841.04  16812145.24  19947739.70   
Saint Petersburg (Pulkovo)  12285537.00  15045980.00  16853137.00   
Ekaterinburg (Koltsovo)      3880241.80   4879549.27   5540180.16   
Sochi                        5024994.36   5379581.11   6029991.85   
Novosibirsk (Tolmachyovo)    3621881.40   4407282.60   5269107.60   
Krasnodar (Paszkowski)       2755624.88   3213341.66   3853329.34   
Ufa                          2120252.96   2570410.28   2983720.31   
Vladivostok (Knevichi)       1686177.23   2004491.19   2430221.39   
Khabarovsk (New)             1652071.01   1895035.45   1980873.25   
Kaliningrad (Hrabrovo)       1459722.69   1667916.01   1998122.16   

                                   2019       2020  True whole year  
Airport name                                                         
Moscow (Sheremetyevo)      

### Выгрузка файла

In [219]:
df2.to_csv('airports.csv')